### PR TITLE
Removing jQuery from Carousel

### PR DESCRIPTION
**Pull request by Johann-S**
Rewrite our Carousel plugin without jQuery

- ~Improve `off` method of our EventHandler to remove listeners in an entire namespace~

Work in progress...

### CODE REVIEW

File: js/src/util.js [hunk 1]
```diff
@@ -88,6 +88,22 @@ const Util = (() => {
           }
         }
       }
+    },
+
+    extend(obj1, obj2) {
```
Comment (wojtask9) [inline js/src/util.js]: Why not use Object.assign?
I thing another issue is that this implementation is `deepCopy` but $.extend by default is not recursive.

Comment (wojtask9) [reply js/src/util.js]: If Object.assign cannot be used I would use `arguments` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) and then you can change `Util.extend(Utils.extend...` to `Util.extend(obj1, obj2, obj3, ...)`

Comment (Johann-S) [reply js/src/util.js]: I didn't had time to think more of this implementation of `extend` but it did the job 😄
I'll take a look thanks @wojtask9 👍 
If you want you can submit a PR which target this branch (`v4-carousel-without-jquery`) with a better implementation

**Comment by Johann-S** [comment]
last issue here, when I call `dispose` the first time all the event listeners were removed, but if click on the `next` arrow (a new carousel is created) and if I call `dispose` again the listeners are still alive 😢

**Comment by Johann-S** [comment]
Now I have to be able to trigger events with parameters inside to be able to run our unit test for our Carousel plugins

**Comment by Johann-S** [comment]
All test passed except one from our Button plugin but it's a known issue

**Pull request merged**